csharp/FAudio.cs

```diff
@@ -39,15 +41,30 @@ public static class FAudio
 
 	#region UTF8 Marshaling
 
-	private static byte[] UTF8_ToNative(string s)
+	private static int Utf8Size(string str)
+	{
+		return (str.Length + 1) * 4;
```

Comment:
Should be `(str.Length * 4) + 1`. I'm a bit nervous about this being on the stack though (16k worst-case, ack!), does AllocHGlobal work here? I'm okay with adding all the FreeHGlobals if performance stays the same here.

Reply:
The use cases are small identifiers and paths, right?

Reply:
Paths are the typical case, and that’s what makes it scary - Win32 MAX_PATH is pretty light but Windows 10 has started extending that number, and on other targets the POSIX PATH_MAX can be anywhere from 1024 all the way to 4096, with NTFS/ext4/etc supporting up to 32678 characters. I suspect this number will only grow rather than shrink in the future.

Reply:
I'll push a commit that switches it over.